test/tower_bugsnag_test.exs

```diff
@@ -231,6 +231,60 @@ defmodule TowerBugsnagTest do
     end)
   end
 
+  test "reports throw with Bandit", %{bypass: bypass} do
+    # An ephemeral port hopefully not being in the host running this code
+    plug_port = 51111
+    url = "http://127.0.0.1:#{plug_port}/uncaught-throw"
+
+    waiting_for(fn done ->
+      Bypass.expect_once(bypass, "POST", "/", fn conn ->
+        {:ok, body, conn} = Plug.Conn.read_body(conn)
+
+        assert(
+          {
+            :ok,
+            %{
+              "events" => [
+                %{
+                  "exceptions" => [
+                    %{
+                      # An exit instead of a throw because Bandit doesn't handle throw's
+                      # for the moment. See: https://github.com/mtrudel/bandit/pull/396.
+                      "errorClass" => "(exit) bad return value: \"something\"",
+                      "message" => "bad return value: \"something\"",
```

Comment:
After digging a bit deeper on why a throw with Bandit end up being a "bad return value" exit, realized that it's because given Bandit doesn't handle throws, and that the plug call runs inside a `:gen_server`, the thrown value gets interpreted as the `:gen_server` callback return value. See https://github.com/mtrudel/bandit/issues/410.

So given that we're testing with `"something"` string here, and that is an invalid `:gen_server` return value, we do get termination of the gen_server and an exit event.

However, if the thrown value were to coincide with a valid `:gen_server` return value, no event would be raised. Extremely unlikely I guess, but technically possible.